integrations/pinecone/tests/test_document_store.py

```diff
@@ -0,0 +1,90 @@
+import time
+from unittest.mock import patch
+
+from haystack import Document
+
+from pinecone_haystack.document_store import PineconeDocumentStore
+
+
+class TestDocumentStore:
+    @patch("pinecone_haystack.document_store.pinecone")
+    def test_init(self, mock_pinecone):
+        mock_pinecone.Index.return_value.describe_index_stats.return_value = {"dimension": 30}
+
+        document_store = PineconeDocumentStore(
+            api_key="fake-api-key",
+            environment="gcp-starter",
+            index="my_index",
+            namespace="test",
+            batch_size=50,
+            dimension=30,
+            metric="euclidean",
+        )
+
+        assert document_store.environment == "gcp-starter"
+        assert document_store.index == "my_index"
+        assert document_store.namespace == "test"
+        assert document_store.batch_size == 50
+        assert document_store.dimension == 30
+        assert document_store.index_creation_kwargs == {"metric": "euclidean"}
```

Comment:
why not checking the api key? I would also add a test to ensure the fallback on the environment variable works